pkg/controllers/applicationfailover/common.go

```diff
@@ -117,3 +125,106 @@ func distinguishUnhealthyClustersWithOthers(aggregatedStatusItems []workv1alpha2
 
 	return unhealthyClusters, others
 }
+
+func buildPreservedLabelState(statePreservation *policyv1alpha1.StatePreservation, rawStatus []byte) (map[string]string, error) {
+	results := make(map[string]string, len(statePreservation.Rules))
+	for _, rule := range statePreservation.Rules {
+		value, err := parseJSONValue(rawStatus, rule.JSONPath)
+		if err != nil {
+			klog.Errorf("Failed to parse value with jsonPath(%s) from status(%v), error: %v",
+				rule.JSONPath, string(rawStatus), err)
+			return nil, err
+		}
+		results[rule.AliasLabelName] = value
+	}
+
+	return results, nil
+}
+
+func parseJSONValue(rawStatus []byte, jsonPath string) (string, error) {
+	template := jsonPath
+	j := jsonpath.New(jsonPath)
+	j.AllowMissingKeys(false)
+	err := j.Parse(template)
+	if err != nil {
+		return "", err
+	}
+
+	buf := new(bytes.Buffer)
+	unmarshalled := make(map[string]interface{})
+	_ = json.Unmarshal(rawStatus, &unmarshalled)
+	err = j.Execute(buf, unmarshalled)
+	if err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func findTargetStatusItemByCluster(aggregatedStatusItems []workv1alpha2.AggregatedStatusItem, cluster string) (workv1alpha2.AggregatedStatusItem, bool) {
+	if len(aggregatedStatusItems) == 0 {
+		return workv1alpha2.AggregatedStatusItem{}, false
+	}
+
+	for index, statusItem := range aggregatedStatusItems {
+		if statusItem.ClusterName == cluster {
+			return aggregatedStatusItems[index], true
+		}
+	}
+
+	return workv1alpha2.AggregatedStatusItem{}, false
+}
+
+func getClusterNamesFromTargetClusters(targetClusters []workv1alpha2.TargetCluster) []string {
+	if targetClusters == nil {
+		return nil
+	}
+
+	clusters := make([]string, 0, len(targetClusters))
+	for _, targetCluster := range targetClusters {
+		clusters = append(clusters, targetCluster.Name)
+	}
+	return clusters
+}
+
+func buildTaskOptions(failoverBehavior *policyv1alpha1.ApplicationFailoverBehavior, aggregatedStatus []workv1alpha2.AggregatedStatusItem, cluster, producer string, clustersBeforeFailover []string) ([]workv1alpha2.Option, error) {
+	var taskOpts []workv1alpha2.Option
+	taskOpts = append(taskOpts, workv1alpha2.WithProducer(producer))
+	taskOpts = append(taskOpts, workv1alpha2.WithReason(workv1alpha2.EvictionReasonApplicationFailure))
+	taskOpts = append(taskOpts, workv1alpha2.WithPurgeMode(failoverBehavior.PurgeMode))
+
+	if failoverBehavior.StatePreservation != nil && len(failoverBehavior.StatePreservation.Rules) != 0 {
+		targetStatusItem, exist := findTargetStatusItemByCluster(aggregatedStatus, cluster)
+		if !exist || targetStatusItem.Status == nil || targetStatusItem.Status.Raw == nil {
+			return nil, fmt.Errorf("the binding status under Cluster(%s) has not been colloected yet ", cluster)
```

Comment:
```suggestion
			return nil, fmt.Errorf("the application status has not yet been collected from cluster: %s", cluster)
```
Fixed typo `colloected`.

Reply:
Updated, thx~